### PR TITLE
Missing loading state when the preview is pending re render

### PR DESCRIPTION
**Pull request by yyorkchan**
Demo:
https://github.com/user-attachments/assets/7a49e645-3f9a-4b7b-a71e-8bbe033c6b3f

- Now Design Screen theme customisation changes synchronously 
- Select Preview Page shows loading

### CODE REVIEW

File: authui/src/inline-preview.ts [hunk 1]
```diff
@@ -15,6 +16,7 @@ function parsePreviewCustomisationMessage(
     return null;
   }
   return {
+    theme: message.theme ?? {},
```
Comment (cychiuae) [inline authui/src/inline-preview.ts:19]: The fallback value should have the same type as the `message.theme`

File: authui/src/inline-preview.ts [hunk 2]
```diff
@@ -89,6 +91,11 @@ export class InlinePreviewController extends Controller {
       outlet?.setValue(value);
     }
 
+    const currentTheme = el.getAttribute("class");
+    if (currentTheme !== customisationMessage.theme) {
+      el.setAttribute("class", customisationMessage.theme);
+    }
```
Comment (cychiuae) [inline authui/src/inline-preview.ts:97]: Here you will also remove other classes. Please just add/more remove the class we are targeting.

File: portal/src/graphql/portal/DesignScreen/DesignScreen.tsx [hunk 3]
```diff
@@ -797,6 +797,8 @@ const Preview: React.VFC<PreviewProps> = function Preview(props) {
 
   const authUIIframeRef = useRef<HTMLIFrameElement | null>(null);
 
+  const [isIframeLoading, setisIframeLoading] = useState(true);
```
Comment (cychiuae) [inline portal/src/graphql/portal/DesignScreen/DesignScreen.tsx:800]: `setisIframeLoading` -> `setIsIframeLoading`

File: portal/src/graphql/portal/DesignScreen/DesignScreen.tsx [hunk 4]
```diff
@@ -827,32 +829,33 @@ const Preview: React.VFC<PreviewProps> = function Preview(props) {
...
     const message = mapDesignFormStateToPreviewCustomisationMessage(
       designForm.state
     );
+    setisIframeLoading(false);
     authUIIframeRef.current?.contentWindow?.postMessage(message, "*");
   }, [designForm.state]);
 
   return (
     <div className={cn("flex", "flex-col", className)}>
+      {isIframeLoading ? <ShowLoading /> : null}
```
Comment (cychiuae) [inline portal/src/graphql/portal/DesignScreen/DesignScreen.tsx:858]: Please pay more attention to the ui you introduce

Screenshot 2024-08-16 at 6 06 10 PM

**Review by cychiuae — CHANGES_REQUESTED** [review]

File: authui/src/inline-preview.ts [hunk 5]
```diff
@@ -89,6 +91,9 @@ export class InlinePreviewController extends Controller {
       outlet?.setValue(value);
     }
 
+    el.classList.remove("light", "dark", "auto");
+    el.setAttribute("class", customisationMessage.theme);
```
Comment (louischan-oursky) [inline authui/src/inline-preview.ts:95]: If you look at authui/src/colorscheme.ts, you will see there is no `light` or `auto`. You have to study how light/dark works in Auth UI first. You never guess.

**Review by louischan-oursky — CHANGES_REQUESTED** [review]

**Review by louischan-oursky — APPROVED** [review]

File: portal/src/graphql/portal/DesignScreen/DesignScreen.tsx [hunk 6]
```diff
@@ -834,6 +834,10 @@ const Preview: React.VFC<PreviewProps> = function Preview(props) {
     []
   );
 
+  useEffect(() => {
+    setIsIframeLoading(true);
+  }, [designForm.state.selectedLanguage]);
+
```
Comment (cychiuae) [inline portal/src/graphql/portal/DesignScreen/DesignScreen.tsx:840]: I suggest just add an effect to set loading to true whenever src is changed

**Review by louischan-oursky — APPROVED** [review]

**Pull request merged**